src/test/regress/expected/rpt.out

```diff
@@ -1329,6 +1353,65 @@ select c from rep_tab where c in (select distinct d from rand_tab);
  2
 (2 rows)
 
+--
+-- Check that nested sub-selects from distributed replicated tables are pulled up if they contain volatiles
+--
+drop table if exists t;
+NOTICE:  table "t" does not exist, skipping
+create table t (i int) distributed replicated;
+insert into t select 1;
+create or replace function f(i int) returns int language sql security definer as $$ select i; $$;
+explain (costs off, verbose) SELECT (select f(i) from t);
+                     QUERY PLAN                      
+-----------------------------------------------------
+ Result
+   Output: $0
+   InitPlan 1 (returns $0)  (slice2)
+     ->  Gather Motion 1:1  (slice1; segments: 1)
+           Output: (f(i))
+           ->  Seq Scan on rpt.t
+                 Output: f(i)
+ Optimizer: Postgres query optimizer
+ Settings: enable_bitmapscan=off, enable_seqscan=off
+(9 rows)
+
+explain (costs off, verbose) SELECT (select f(i) from t group by f(i));
+                     QUERY PLAN                      
+-----------------------------------------------------
+ Result
+   Output: $0
+   InitPlan 1 (returns $0)  (slice2)
+     ->  Gather Motion 1:1  (slice1; segments: 1)
+           Output: (f(i))
+           ->  HashAggregate
+                 Output: (f(i))
+                 Group Key: f(t.i)
+                 ->  Seq Scan on rpt.t
+                       Output: i, f(i)
+ Optimizer: Postgres query optimizer
+ Settings: enable_bitmapscan=off, enable_seqscan=off
+(12 rows)
+
+explain (costs off, verbose) SELECT (select i from t group by i having f(i) > 0);
+                     QUERY PLAN                      
+-----------------------------------------------------
+ Result
+   Output: $0
+   InitPlan 1 (returns $0)  (slice2)
+     ->  Gather Motion 1:1  (slice1; segments: 1)
+           Output: i
```

Comment:
@RekGRpth, can you explain how this patch makes this plan correct? At first glance, there is no volatile function in the target list.

Reply:
Since `make_subplan` does not set `CdbLocusType_SingleQE` and `FLOW_SINGLETON` for `CdbLocusType_SegmentGeneral`, then in `ParallelizeSubplan` value of `containingPlanDistributed` became false and plan is focused.

Reply:
containingPlanDistributed is calculated for outer for subplan plan. `subPlanDistributed` described subplan distribution.